src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitMemberAccessor.cs

```diff
@@ -7,12 +7,27 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Reflection;
 using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
 
 namespace System.Text.Json.Serialization.Metadata
 {
     [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
     internal sealed class ReflectionEmitMemberAccessor : MemberAccessor
     {
+        private const int MaxInterpretedConstructorParameters = 64;
+        private static ReflectionMemberAccessor? s_reflectionMemberAccessor;
+        private readonly bool _isDynamicCodeInterpreted;
```

Comment:
```suggestion
        private static readonly bool s_isDynamicCodeInterpreted;
```

This isn't going to change for the lifetime of the app.

Reply:
I'd also suggest naming this the same as `RuntimeFeature.IsDynamicCodeCompiled` to avoid any confusion.

```suggestion
        private static readonly bool s_isDynamicCodeCompiled;
```

Reply:
The constructor contains a debug assert, so wanted to avoid using a static ctor here.

Reply:
You can do that with

```C#
private static readonly bool s_isDynamicCodeCompiled = InitializeIsDynamicCodeCompiled();

private static bool InitializeIsDynamicCodeCompiled()
{
#if NETCOREAPP
            Debug.Assert(RuntimeFeature.IsDynamicCodeSupported);
            _isDynamicCodeInterpreted = !RuntimeFeature.IsDynamicCodeCompiled;
#else
            _isDynamicCodeInterpreted = false;
#endif
}
```

Reply:
That would still be invoke by a static constructor though, so any potential failures would happen nondeterministically at type initialization. It shouldn't matter much though; the member accessor is initialized as a singleton:

https://github.com/dotnet/runtime/blob/38ce8bb2422e14c142254a72f02382dfa5b644c7/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L595-L606

Reply:
Caching `RuntimeFeature.IsDynamicCodeSupported` in a variable is a de-optimization. This is one of those properties that gets inlined by linker or JIT and the code is dead-code eliminated when called directly. If you cache it in a variable, none of those optimizations kick in.

Can you just call it directly without caching?

Reply:
> Caching RuntimeFeature.IsDynamicCodeSupported in a variable is a de-optimization. 

These methods are only called once per member during metadata initialization, so it shouldn't matter much. Because the property is not available in .NET framework, resolving to a field removes the need for build conditionals inside the method logic itself.

Reply:
You need to have an ifdef somewhere. If you do not like the ifdef in the middle of the method, you can make this a (static) property instead of field.